src/DeviceActivity.cs

```diff
@@ -0,0 +1,54 @@
+using Newtonsoft.Json.Linq;
+using System.Collections.Generic;
+
+namespace Razorpay.Api
+{
+    public class DeviceActivity : Entity 
+    {
+        public DeviceActivity(string activityId = "")
+        {
+            this["id"] = activityId;
+        }
+
+        /// <summary>
+        /// Create a new device activity for POS gateway
+        /// </summary>
+        /// <param name="data">Dictionary containing device activity data in the format expected by rzp-pos-gateway</param>
+        /// <returns>DeviceActivity object</returns>
+        public DeviceActivity Create(Dictionary<string, object> data)
+        {
+            string relativeUrl = string.Format("{0}/{1}", GetUrlVersion(), GetEntityUrl());
+            List<Entity> entities = Request(relativeUrl, HttpMethod.POST, data, "POS");
+            return (DeviceActivity)entities[0];
+        }
+
+        /// <summary>
+        /// Get the status of a device activity
+        /// </summary>
+        /// <param name="activityId">Activity ID to fetch status for</param>
+        /// <returns>DeviceActivity object with current status</returns>
+        public DeviceActivity GetStatus(string activityId)
+        {
+            if (string.IsNullOrWhiteSpace(activityId))
+            {
+                throw new System.ArgumentException("Activity ID cannot be null or empty", "activityId");
```

Comment:
this is not standard error format, either error should be standard error or remove the validation